pystac/collection.py

```diff
@@ -715,7 +715,9 @@ def from_dict(
 
         if assets is not None:
             for asset_key, asset_dict in assets.items():
-                collection.add_asset(asset_key, Asset(asset_dict))
+                asset_dict = dict(asset_dict)
+                asset_dict.setdefault("media_type", asset_dict.pop("type", None))
```

Comment:
I'll assume it's deliberate that we use `media_type` here instead of `type`. Anything else I should watch out for?